R/recode-with-table.R

```diff
@@ -0,0 +1,1033 @@
+#' Recodes columns from passed row and returns just table with those columns
+#' and same rows as the data
+#'
+#' @param data The source database
+#' @param variables_to_process rows from variable details that are applicable
+#' to this DB
+#' @param data_name Name of the database being passed
+#' @param log The option of printing log
+#' @param print_note the option of printing the note columns
+#' @param else_default default else value to use if no else is present
+#'
+#' @return Returns recoded and labeled data
+recode_columns <-
+  function(data,
+           variables_to_process,
+           data_name,
+           log,
+           print_note,
+           else_default) {
+    # Split variables to process into recode map and func
+    map_variables_to_process <-
+      variables_to_process[grepl("map::", variables_to_process[[pkg.env$columns.CatValue]]),]
+
+    id_variables_to_process <-
+      variables_to_process[grepl("id_from::", variables_to_process[[pkg.env$columns.CatValue]]),]
+
+    func_variables_to_process <-
+      variables_to_process[grepl("Func::", variables_to_process[[pkg.env$columns.CatValue]]),]
+
+    rec_variables_to_process <-
+      variables_to_process[(!grepl("Func::|map::|id_from::", variables_to_process[[pkg.env$columns.CatValue]])) &
+                             (!grepl("DerivedVar::",
+                                     variables_to_process[[pkg.env$columns.VariableStart]])),]
+
+    label_list <- list()
+    # Set interval if none is present
+    valid_intervals <- c("[,]", "[,)", "(,]")
+    interval_default <- "[,]"
+    recoded_data <- data[, 0]
+
+    # Loop through the rows of recode vars
+    while (nrow(rec_variables_to_process) > 0) {
+      variable_being_checked <-
+        as.character(rec_variables_to_process[1,
+                                              pkg.env$columns.Variables])
+      rows_being_checked <-
+        rec_variables_to_process[rec_variables_to_process[[pkg.env$columns.Variables]] == variable_being_checked,]
+      rec_variables_to_process <-
+        rec_variables_to_process[!rec_variables_to_process[[pkg.env$columns.Variables]] == variable_being_checked,]
+      first_row <- rows_being_checked[1,]
+      # Check for varialbe existance in data
+      data_variable_being_checked <-
+        get_data_variable_name(
+          data_name = data_name,
+          row_being_checked = first_row,
+          variable_being_checked = variable_being_checked,
+          data = data
+        )
+      if (is.null(data[[data_variable_being_checked]])) {
+        warning(
+          paste(
+            "Data",
+            data_name,
+            "does not contain the variable",
+            data_variable_being_checked
+          )
+        )
+      } else {
+        # Check for From column duplicates
+        all_from_values_for_variable <-
+          rows_being_checked[[pkg.env$columns.From]]
+        if (length(unique(all_from_values_for_variable)) != length(all_from_values_for_variable)) {
```

Comment:
Yeah its checking for duplicates

Reply:
I haven't tested this but if the recFrom values are  [1,10], [2,8] does it catch that its an error?

Reply:
Nope its for if you have 2 rows with the same range such as variable details row 1 has range of [1,10] recoded to 5 and row 2 has [1,10] recoded to 6 since without this catch it will just recode to last row and this unpredictable so duplicate identical ranges for the same variable are not allowed

Reply:
Sorry, I'm confused, does it catch the scenario I mentioned above? If not it should, because we don't ambigous conditions in recFrom.